pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -460,24 +461,59 @@ func (metrics *vmiMetrics) updateNetwork(vmStats *stats.DomainStats) {
 	}
 }
 
-func makeVMIsPhasesMap(vmis []*k6tv1.VirtualMachineInstance) map[string]uint64 {
-	phasesMap := make(map[string]uint64)
+type vmiCountMetric struct {
+	phase    string
+	Os       string
+	Workload string
+	Flavor   string
+}
 
-	for _, vmi := range vmis {
-		phasesMap[strings.ToLower(string(vmi.Status.Phase))] += 1
+func (vmc *vmiCountMetric) SetField(name string, value string) {
+	f := reflect.ValueOf(vmc).Elem().FieldByName(strings.Title(name))
+	if f.IsValid() && f.CanSet() {
+		f.SetString(value)
 	}
+}
+
+func (vmc *vmiCountMetric) UpdateFromAnnotations(annotations map[string]string) {
+	t := reflect.TypeOf(*vmc)
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		if val, ok := annotations["vm.kubevirt.io/"+strings.ToLower(field.Name)]; ok {
```

Comment:
do we have the `vm.kubevirt.io` as a const anywhere or how is this value used/added to the annotations in the vmi template ?

Reply:
it's added in common-templates https://github.com/kubevirt/common-templates/pull/288